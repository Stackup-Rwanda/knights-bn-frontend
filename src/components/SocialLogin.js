import React from 'react';
import { FcGoogle } from 'react-icons/fc';
import { FaFacebookF } from 'react-icons/fa';
import SocialLoginIcon from './SocialLoginIcon';
import '../assets/styles/containers/loginPage.scss';

const SocialLogin = ({ googleAction, facebookAction }) => (
  <div>
    <SocialLoginIcon
      icon={FcGoogle}
      label={googleAction}
      class="mr-2 color"
      formClass="social-login-btn"
      buttonClass="btnStyle"
      action="https://knights-bn-backnd.herokuapp.com/api/v1/auth/login/google"
    />
    <SocialLoginIcon
      icon={FaFacebookF}
      label={facebookAction}
      class="mr-2 ml-3 text-light bg-info"
      formClass="social-login-btn"
      buttonClass="btnStyle"
      action="https://knights-bn-backnd.herokuapp.com/api/v1/auth/login/facebook"
    />
  </div>
);

export default SocialLogin;
