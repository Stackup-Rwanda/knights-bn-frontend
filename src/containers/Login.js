import React, { Component } from 'react';
import '../assets/styles/containers/landingPage.scss';


class Login extends Component {
  render() {
    return (
      <div className="background">
        <h2>Barefoot Nomad Login Page</h2>

      </div>
    );
  }
}
export default Login;
