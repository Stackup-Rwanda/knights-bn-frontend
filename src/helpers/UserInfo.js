import React, { Component } from 'react';
import queryString from 'query-string';

export default class UserInfo extends Component {
  componentDidMount() {
    try {
      const userInfo = queryString.parse(this.props.location.search);
      const { data } = userInfo;
      const userData = JSON.parse(data);
      const { user, token } = userData;
      localStorage.setItem('user-token', token);
      localStorage.setItem('user-info', JSON.stringify(user));
      window.location.href = '/home';
    } catch (error) {
      window.location.href = '/lost';
    }
  }

  render() {
    return (
      <div>
        <h1>Request loading error</h1>
      </div>
    );
  }
}
