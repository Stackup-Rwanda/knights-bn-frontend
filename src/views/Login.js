import React, { Component } from 'react';
import { Form, Spinner } from 'react-bootstrap';
import '../assets/styles/containers/loginPage.scss';
import { connect } from 'react-redux';
import {
  fetchUsers,
  validateInput,
  loginUsersSuccess,
  handleError,
  setLoadingStatus,
} from '../redux/actions/index';
import { Redirect } from 'react-router-dom';
import TextBox from '../components/Textbox';
import Button from '../components/Button';
import Line from '../components/line';
import { thunk } from '../redux/thunk/index';
import { withRouter } from 'react-router';
import AlertComponent from '../components/Alert';
import SocialLogin from '../components/SocialLogin';

export class Login extends Component {
  state = {
    data: { email: '', password: '' },
    error: ' ',
    loggedIn: false,
  };

  login = async (e) => {
    e.preventDefault();
    this.props.handleError('');
    const { validations: validatedFields } = this.props;
    const invalidFields = Object.keys(validatedFields).filter(
      (field) => validatedFields[field] === 'is-invalid'
    );
    let validatedFieldsToArray = Object.keys(validatedFields);
    if (validatedFieldsToArray.length > 1 && invalidFields.length === 0) {
      this.props.setLoadingStatus(true);
      await this.props.thunk(
        'post',
        '/auth/login',
        loginUsersSuccess,
        this.props.user
      );
      this.props.setLoadingStatus(false);

      const { isLoggedIn } = this.props.userData;
      isLoggedIn
        ? this.props.history.push('/home')
        : this.props.handleError(
            'Incorrect email or password. Please try again.'
          );
    } else {
      const errorMsg =
        validatedFieldsToArray.length === 0
          ? 'Please fill in the required fields.'
          : this.props.validations.email === 'is-valid' &&
            !this.props.validations.password
          ? 'Please enter your password.'
          : 'Wrong email address.';
      this.props.handleError(errorMsg);
    }
  };

  render() {
    const { validateInput, error, isLoading } = this.props;
    return (
      <div className="login-page loginContainer">
        <Spinner
          animation="border"
          variant="primary"
          className={isLoading ? 'spinner--position__center' : 'hide'}
        />
        <Form className="form">
          <h1>Sign in</h1>
          <AlertComponent isError={error ? true : false} errorMsg={error} />
          <p>Login with social media</p>
          <SocialLogin
            googleAction="Sign in with Google"
            facebookAction="Sign in with Facebook"
          />
          <div id="or">
            <Line className="Line" />
            <div id="or_">
              <span>or</span>
            </div>
            <Line className="Line_" />
          </div>
          <TextBox
            type="email"
            placeholder="email"
            id="email"
            name="email"
            onChange={validateInput}
            label="email"
            value={this.props.user.email}
          />
          <TextBox
            type="password"
            placeholder="password"
            id="password"
            name="password"
            onChange={validateInput}
            label="password"
            value={this.props.user.password}
          />
          <p>{this.props.token ? <Redirect to="home" /> : null}</p>
          <Button
            aria-label="login"
            onClick={(e) => this.login(e)}
            id="loginBtn"
            label="Login"
          />
          <a id="forgotPassword" href="#">
            <p>Forgot Password?</p>
          </a>
        </Form>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    token: state.user.token,
    error: state.errorHandler.error,
    validations: state.eventHandler.validations,
    user: state.eventHandler.user,
    userData: state.user.data,
    isLoading: state.eventHandler.isLoading,
  };
};

const mapDispatchToProps = {
  fetchUsers,
  validateInput,
  thunk,
  handleError,
  setLoadingStatus,
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Login));
