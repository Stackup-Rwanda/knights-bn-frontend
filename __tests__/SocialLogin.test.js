import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount, shallow, render } from '../setupTests';
import SocialLogin from '../src/components/SocialLogin';
import UserInfo from '../src/helpers/UserInfo';
import Login from '../src/views/Login';

describe('testing Social media', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Login debug />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<SocialLogin debug />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly in "debug" mode', () => {
    const component = shallow(<UserInfo debug />);
    expect(component).toMatchSnapshot();
  });
  it('should render banner text correctly with given strings', () => {
    const props = {
      location: {
        pathname: '/userinfo/',
        search:
          '?data=%7B"user":%7B"id":12,"firstName":"Niyonsenga","lastName":"Eric","gender":"male","email":"niyeric11@gmail.com","passport":"12345678","password":"$2b$10$Auc8QIr7B26kMEikqxvlOOzBx3XNTi3sNjadNreP5nc19A4D2n0H.","method":null,"clientId":null,"lineManager":null,"birthDay":null,"language":null,"currency":null,"homeTown":null,"role":null,"department":null,"biography":null,"profileImage":null,"createdAt":"2020-04-18T10:06:42.206Z","updatedAt":"2020-04-18T10:09:34.053Z"%7D,"token":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTIsImVtYWlsIjoibml5ZXJpYzExQGdtYWlsLmNvbSIsImZpcnN0TmFtZSI6Ik5peW9uc2VuZ2EiLCJsYXN0TmFtZSI6IkVyaWMiLCJyb2xlIjpudWxsLCJpYXQiOjE1ODg1ODg5ODd9.vGGFnCGyw-J8sj-D5ZaeqeZU8dBZM3SwVpX9EhlRmYk"%7D',
        hash: '',
        state: undefined,
      },
    };
    const component = shallow(<UserInfo {...props} />);
    expect(component.text().includes('Request loading error')).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
