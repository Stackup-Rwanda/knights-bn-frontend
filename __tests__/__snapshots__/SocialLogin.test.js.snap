// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Social media should render banner text correctly with given strings 1`] = `
<div>
  <h1>
    Request loading error
  </h1>
</div>
`;

exports[`testing Social media should render correctly in "debug" mode 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`testing Social media should render correctly in "debug" mode 2`] = `
<div>
  <SocialLoginIcon
    action="https://knights-bn-backnd.herokuapp.com/api/v1/auth/login/google"
    buttonClass="btnStyle "
    class="mr-2 "
    formClass="social-login-btn"
    icon={[Function]}
  />
  <SocialLoginIcon
    action="https://knights-bn-backnd.herokuapp.com/api/v1/auth/login/facebook"
    buttonClass="btnStyle"
    class="mr-2 ml-3 text-light SocialFacebook"
    formClass="social-login-btn"
    icon={[Function]}
  />
</div>
`;

exports[`testing Social media should render correctly in "debug" mode 3`] = `
<div>
  <h1>
    Request loading error
  </h1>
</div>
`;
